src/common/config.cc

```diff
@@ -1227,6 +1227,47 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other,
+            std::map<std::string, std::pair<std::string, std::string>> *diff, 
+            std::set<std::string> *unknown, const std::string& ceph_setting, bool show_unchanged)
+{
+  Mutex::Locker l(lock);
+
+  bool wasItFound = false;
+  char local_buf[4096];
+  char other_buf[4096];
+  for (int i = 0; i < NUM_CONFIG_OPTIONS; i++) {
+    if (wasItFound) break;  //-- If we already found it, we are done.
+    config_option *opt = &config_optionsp[i];
+    std::string ceph_setting_name(opt->name);
+    if (ceph_setting == ceph_setting_name) {
+      memset(local_buf, 0, sizeof(local_buf));
+      memset(other_buf, 0, sizeof(other_buf));
+
+      char *other_val = other_buf;
+      int err = other->get_val(opt->name, &other_val, sizeof(other_buf));
+      if (err < 0) {
+        if (err == -ENOENT) {
+          unknown->insert(opt->name);
+        } //-- if (err == -ENOENT)
+        continue;
+      } //-- if (err < 0)
+
+      char *local_val = local_buf;
+      err = _get_val(opt->name, &local_val, sizeof(local_buf));
+      if (err != 0) continue;
+      if (strcmp(local_val, other_val)) diff->insert(make_pair(opt->name, make_pair(local_val, other_val)));
+      else {
+        if (show_unchanged) diff->insert(make_pair(opt->name, make_pair(local_val, other_val)));
```

Comment:
I have not seen any cases where config_optionsp[] would have duplicated values. If so, we would need to think about changing the array in question to a hash (as I suggested during the  work done for  http://tracker.ceph.com/issues/14934).